image/data.go

```diff
@@ -0,0 +1,59 @@
+package image
+
+type Images struct {
+	Count      int     `json:"count"`
+	Items      []Image `json:"items"`
+	Limit      int     `json:"limit"`
+	Offset     int     `json:"offset"`
+	TotalCount int     `json:"total_count"`
+}
+
+type NewImage struct {
+	CollectionId string  `json:"collection_id,omitempty"`
+	Filename string  `json:"filename,omitempty"`
+	License License   `json:"license,omitempty"`
+	Type string  `json:"type,omitempty"`
+}
+
+type License struct {
+	Title string  `json:"title,omitempty"`
+	Href string  `json:"href,omitempty"`
+}
+
+type Image struct {
+	Id string  `json:"id,omitempty"`
+	CollectionId	 string  `json:"collection_id,omitempty"`
+	State string  `json:"state,omitempty"`
+	//enum:
+	//- created
+	//- uploaded
+	//- publishing
+	//- published
+	//- deleted
```

Comment:
I am not sure if anything should be done with enums in clients. I have temporarily left the comments in place for discussion in this PR.

Reply:
In dataset client, I used iota constants for the state enumeration. Maybe we can do something similar?:
```
// Possible values for a State of the resource. It can only be one of the following:
// TODO these states should be enforced in all the 'POST' and 'PUT' operations that can modify states of resources
const (
	StateCreated State = iota
	StateUploaded
	StatePublishing
	StatePublished
	StateDeleted
)

var stateValues = []string{"created", "uploaded", "publishing", "published", "deleted"}

// String returns the string representation of a state
func (s State) String() string {
	return stateValues[s]
}
```

Reply:
I did state validation in the server, I think I should not block this PR because of this reason - I will approve it.